### ISSUE TITLE
Amount specification in instructions does not scale with recipe

### ISSUE THREAD

**Issue by AberDerBart** [tstehr/recipemd#1]
When multiplying the yield of a recipe with the -m or -s options, the amounts given in the instructions do not scale.

I suggest writing them in italic for easy parsing and emphasis in the rendered recipe.

**Comment by tstehr** [comment]
Agreed, this is a commonly requested feature and its absence currently requires workarounds:

- State the ingredient multiple times in the ingredient list, differentiated by use (e.g. `*2* spring onions for base` and `*2* spring onions for garnish`)
- Use fractions of ingredients in the instructions (e.g. `Add half the spring onions`) 

However, the two obvious approaches to do this are both bad for different reasons: 

1. Try to parse all text in emphasis as an amount, ignore parsing fails
   - This creates a minimal amount of breakage for existing recipes
   - It keeps emphasis available for non-amount usage
   - It creates a risk of recipes containing subtly malformed amounts in instructions (e.g. one of the cases mentioned in https://github.com/tstehr/RecipeMD/issues/38#issuecomment-1368443234). The user would not be notified and scaling would create invalid recipes

2. Parse all text in emphasis as an amount, fail recipe parsing if the text is not a valid amount
   - Breaking change, would require new major version and possibly a migration tool
   - Consumes emphasis for amount usage. In particular once we fix #38 this means you could no longer write stuff like `Fold in egg whites *very carefully* until *just* incorporated` in the instructions, since `very carefully` and `just` are not valid amounts

Overall I personally slightly prefer option 1, but I'm not very happy with it. 

I would be very interested in input for alternative solutions.

**Comment by tstehr** [comment]
> The risk for the first option can be simply taken care of by calculating, if the ingredients in the instructions match the ingredient list. Overall this is my personal benefit of this issue (being able to detect missing ingredients in the ingredient list).

I'm not sure I fully understand what you want to do here. Would you expect every ingredient's amount to be restated verbatim in the recipe's instructions? How would you match ingredient amounts with the instruction amounts, given that you can have multiple ingredients with the same amount?

Personally I never spell out the amount in the instructions if it's exactly equal to the amount already given in the ingredient list. I'd only add amount in the instructions where they differ from the amount in the ingredient list (e.g. the same ingredient is used in two ways) or for ingredients that don't need to show up in the ingredient list (special case, I would probably only use it for "tap water").

For clarity, I interpret such that only the amount, not the full ingredient is written in italics. Example recipe:

```markdown
# Tamago Kake Gohan 
---

- *340 g* cooked white rice
- *2* large eggs
- *1/2 teaspoon* soy sauce

--- 

- Place rice in a bowl and make a shallow indentation in the center. Break *1* whole egg into the center. 
- Season with *1/2 teaspoon* soy sauce.
- Stir vigorously with chopsticks to incorporate egg; it should become pale yellow, frothy, and fluffy in texture. 
- Taste and adjust seasonings as necessary.
- Make a small indentation in the top, and add  yolk of the other egg. Serve immediately.
```

**Comment by tstehr** [comment]
>  Overall this is my personal benefit of this issue (being able to detect missing ingredients in the ingredient list).

> I'm not sure I fully understand what you want to do here. Would you expect every ingredient's amount to be restated verbatim in the recipe's instructions?
> 
> Yep, that's the idea, which also motivates #49 and #50 . [...]

This is a cool idea, which I would love to see implemented! It would be a great additional tool that recipe authors can optionally use if they like.

However, I think making it a rule/recommendation that you should include all ingredients with their exact amount in the recipe body is too strict. As far as I can see no public recipe RecipeMD collection follows that style. 

I personally wouldn't want to do it because it feels tedious to have to state everything twice. Instead I only state the ingredient name in the instructions, during cooking I can then look up the amount in the ingredient list. For use cases where it is very important that the ingredient list and the instructions always match a language like [cooklang](https://cooklang.org) might be a better match.

My use case for this feature is actually exactly situations where the amounts in the ingredient list and in the instructions do *not* match because an ingredient it used two different ways during the preparation of the recipe (e.g. eggs in my example recipe).

> > How would you match ingredient amounts with the instruction amounts, given that you can have multiple ingredients with the same amount?
> 
> I don't understand the Problem.

Assume a recipe the contains the ingredients `*20 g* foo` and `*20 g* bar*`. With the original proposal the instructions could be something like `Mix *20 g* bar and *20 g* of foo`. Tooling would have no way to understand which amount belongs to which ingredient looking at the amounts alone.

> Always thought of writing the full ingredient in italics. 

This would feel very inconsistent to me. I'd much rather have the rule of: "Write ingredient amounts in italics", than "Write amounts in italics in the ingredient list, and ingredients in with amounts in the recipe instructions".

From a technical perspective, it would lead to parsing problems where we can't find the boundary between amount unit and ingredient name. Take `*1/2 teaspoon soy sauce*` from your example: Is this an ingredient with the amount `1/2` and the name `teaspoon soy sauce` or an ingredient with the amount `1/2 teaspoon` and the name `soy sauce`?

The only way to feasibly solve this it to look back at the ingredient list. However, that would introduce tons of ambiguity and complicate parsing. It would also not work for my "split ingredient" use case and could very easily lead to unexpected behavior.
